### PR TITLE
👌 IMPROVE: Remove deprecated code (#1373)

### LINKED ISSUE #1373: remove unused/deprecated code targeting older versions of WooCommerce
## Is your feature request related to a problem? Please describe.
We've recently [clarified and updated the versions of WooCommerce that are supported for Storefront](https://docs.woocommerce.com/document/storefront-requirements-version-support/). There may be code targeting older versions which is no longer needed. Removing this code could simplify the code base and reduce the amount of code that needs to be shipped & parsed.

## Describe the solution you'd like
- Search code for `WC_VERSION` constant to find relevant code and safely remove as appropriate.
- There may be other ways to find out of date js/css code to remove also.

Follow up issue from discussion in https://github.com/woocommerce/storefront/pull/1368#discussion_r435688925

### PR DESCRIPTION
**Pull request by nielslange**
Fixes #1373

Remove deprecated code.

### How to test the changes in this Pull Request:

1. When running a WooCommerce version < 2.3, the `add_to_cart_fragments` filter will no longer work.
2. When running a WooCommerce version < 2.5, the `star_rating` script will no longer work.
3. When running a WooCommerce version < 3.3, the `products_per_page` script will no longer work.
4. When running a WooCommerce version < 3.3, the `loop_shop_columns` filter and
the `woocommerce_before_shop_loop` and `woocommerce_after_shop_loop` actions will no longer work.

### Changelog

> Dev – Removed code for unsupported versions of WooCommerce (3.3 and earlier) #1373

### CODE REVIEW

**Comment by haszari** [comment]
@nielslange  Can you please flesh out the PR description with a bit more information about risk/impact? 

For example, for each removed feature, can you clarify how it works in supported versions of Woo, and how to test the flow (to confirm the supported flow is still working). Or if the feature itself is deprecated, add details explaining this.

Although this is probably a very safe change, it's a good habit to state clearly what flows are affected (or in this case, should be unaffected), and detail any risks related to code changes in the PR. Note also the changelog should be succinct and informative, e.g. `Dev – Removed code for unsupported versions of WooCommerce (3.3 and earlier)`; ideally include a clue about the affected area (e.g. star rating, pagination etc). 

Also I assume you've tested this all yourself too?

**Comment by nielslange** [comment]
> @nielslange Can you please flesh out the PR description with a bit more information about risk/impact?
> 
> For example, for each removed feature, can you clarify how it works in supported versions of Woo, and how to test the flow (to confirm the supported flow is still working). Or if the feature itself is deprecated, add details explaining this.

As suggested in https://github.com/woocommerce/storefront/issues/1373, I removed all functions related to unsupported versions. These flows could be tested by installing the corresponding version of WooCommerce. Let's take the `add_to_cart_fragments` filter as mentioned in https://github.com/woocommerce/storefront/pull/1471#issue-489672372 as an example. One would have to install WooCommerce 2.2 to check if the filter works without this PR. Then, checking out this PR and testing again if the filter still works. As the latest version of WooCommerce does not support version 2.2, the filter would no longer work. 

> Although this is probably a very safe change, it's a good habit to state clearly what flows are affected (or in this case, should be unaffected), and detail any risks related to code changes in the PR. Note also the changelog should be succinct and informative, e.g. `Dev – Removed code for unsupported versions of WooCommerce (3.3 and earlier)`; ideally include a clue about the affected area (e.g. star rating, pagination etc).

The README note had been updated as suggested.

> Also I assume you've tested this all yourself too?

As I simply removed the conditional checks including the function calls and the function definitions itself, if no longer needed, I assume it's safe to say that this PR has no effect of supported versions. That said, to be 100% certain, this PR would have to be tested against the following versions of WooCommerce:

- 2.2 and 4.5.x to test the removed `add_to_cart_fragments` filter
- 2.4 and 4.5.x to test the removed `star_rating` script
- 3.2 and 4.5.x to test the removed `products_per_page` script
- 3.2 and 4.5.x to test the removed `loop_shop_columns` filter
- 3.2 and 4.5.x to test the removed `woocommerce_before_shop_loop` action
- 3.2 and 4.5.x to test the removed `woocommerce_after_shop_loop` action

Long story short, I haven't tested this PR against version [2.2](https://woocommerce.com/posts/woocommerce-2-2/), [2.4](https://woocommerce.com/posts/woocommerce-2-4-release/) and [3.2](https://woocommerce.com/posts/whats-new-woocommerce-3-2/).

**Comment by haszari** [comment]
Thanks for the info @nielslange 👍 

I'm keen to get this merged and included in 2.8.0. I'll review the removed code and figure out what the relevant features/user flows are, then do some testing to ensure there's no impact on these features (i.e. the new/un-deprecated implementations). I'll document my findings here for transparency/clarity and if all goes well I'll merge 🎉

File: inc/jetpack/class-storefront-jetpack.php [hunk 1]
```diff
@@ -90,10 +90,7 @@ public function jetpack_infinite_scroll_loop() {
 		 *
 		 * @return void
 		 */
-		public function jetpack_infinite_scroll_wrapper_columns() {
-			add_action( 'storefront_jetpack_product_infinite_scroll_before', 'storefront_product_columns_wrapper' );
-			add_action( 'storefront_jetpack_product_infinite_scroll_after', 'storefront_product_columns_wrapper_close' );
-		}
+		public function jetpack_infinite_scroll_wrapper_columns() {}
```
Comment (haszari) [inline inc/jetpack/class-storefront-jetpack.php:93]: What's the deprecated feature here? I don't see a woo version number. 

@nielslange  can you provide more info about this change in the PR description?

If we do end up removing the body of this routine, should remove the whole function too?

File: inc/woocommerce/storefront-woocommerce-template-functions.php [hunk 2]
```diff
@@ -214,18 +201,6 @@ function storefront_loop_columns() {
...
-	 * Product columns wrapper close
-	 *
-	 * @since   2.2.0
-	 * @return  void
-	 */
-	function storefront_product_columns_wrapper_close() {
-		echo '</div>';
-	}
-}
-
```
Comment (haszari) [inline inc/woocommerce/storefront-woocommerce-template-functions.php:228]: Could child themes or other customizations be using these functions (`storefront_product_columns_wrapper` and `storefront_product_columns_wrapper_close`)? If they're available there may be stores that are using them. 

If we intend to remove them we'll need to deprecate and document any changes so stores can adapt their customisations.

(I'm assuming there's low chance of stores using these, but we need to take care just in case.)

**Review by haszari — COMMENTED** [review]
Thanks for tackling this @nielslange ! It will be nice to trim down the amount of code we're running 😄 

Before we merge this, can you identify what [flows/features this code is related to](https://github.com/woocommerce/storefront/wiki/Testing-Storefront:-flows-and-features)? I.e. for a each code change, what Storefront feature is it part of (or Woo or WordPress or Jetpack as appropriate). If it's powering a deprecated feature that's no longer available or relevant, it's still useful to be explicit about what we're removing.

Then we can test to ensure that there's no unexpected impact of these changes. Having this level of info in the PR helps me when I'm reviewing & testing, and also is useful when smoke testing the release. So thanks in advance!

**Comment by budzanowski** [comment]
@nielslange is there any chance 😉 for a follow-up to the questions. I am considering if we should close this a stale.

**Comment by nielslange** [comment]
> @nielslange is there any chance 😉 for a follow-up to the questions. I am considering if we should close this a stale.

Sorry for my late reply. I've been occupied with other tasks for the previous weeks and missed @haszari's comments. I'll go through the feedback the next days. That said, if you want to stale the PR instead, feel free doing so.

**Comment by mikejolley** [comment]
I've moved the approved changes to https://github.com/woocommerce/storefront/pull/1567 and set you as a co-author. Thanks!
